client/trino-client/src/main/java/io/trino/client/SourceBuilder.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.client;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Joiner;
+import com.google.common.base.StandardSystemProperty;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Map;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+
+public class SourceBuilder
+{
+    private static final Joiner.MapJoiner MAP_JOINER = Joiner.on(" ").withKeyValueSeparator("#");
+    private static final String SEPARATOR = "/";
+
+    private static final String METADATA = "md";
+    private static final String LANGUAGE = "lang" + SEPARATOR + "java";
+    private static final String OS = "os";
+    private static final String ARCH = METADATA + SEPARATOR + "arch";
+    private static final String VENDOR = METADATA + SEPARATOR + "vendor";
```

Comment:
We might want to use similar fields in other languages, like in Python we might specify if it's CPython, pypy, IronPython, etc
```suggestion
    private static final String JAVA_VENDOR = METADATA + SEPARATOR + "javaVendor";
```